# STAR Alignment Guide

For an introduction to using the high performance clusters, look {HPC Introduction}`../HPC/introduction.md`

To perform an RNA-seq alignment using the STAR algorithm, you need the following:
- Sequencing data (.fastq)
-   Produced by sequencer
-   ASTQ files contain the sequences frome each read and quality scores estimating the confidence of each base


- Genome index
-  Generated from reference genome
-  Genome sequences against which sequencing reads are aligned with indexes generated by STAR algorithm

For more information, manual is here.

A good introductory reading: [Link](https://hbctraining.github.io/Intro-to-rnaseq-hpc-salmon/lessons/03_QC_STAR_and_Qualimap_run.html)


# Aligning Bulk Sequencing Data
```bash
cd /dir/to/fastq/files
/gpfs/group/jin/software/STAR-2.7.9a/source/STAR --genomeLoad LoadAndExit --genomeDir /gpfs/group/jin/ref_genomes/mm10

mkdir outs

for i in $(ls *R1*.fastq | sed 's/_R1.*//g'); 
    do mkdir outs/${i}
    /gpfs/group/jin/software/STAR-2.7.9a/source/STAR --runThreadN=16 \
    --genomeDir /gpfs/group/jin/ref_genomes/mm10  \
    --sjdbGTFfile /gpfs/group/jin/ref_genomes/mm10/refdata-gex-mm10-2020-A/genes/genes.gtf\
    --readFilesIn ${i}_R2_001.fastq ${i}_R1_001.fastq \
    --readFilesCommand zcat \
    --quantMode GeneCounts \
    --outFileNamePrefix outs/${i}/; done

/gpfs/group/jin/software/STAR-2.7.9a/source/STAR --genomeLoad Remove --genomeDir /gpfs/group/jin/ref_genomes/mm10
```

# Aligning Single Cell Sequencing Data
Alignment needs 10x barcode list. V3 chemistry whitelist is downloaded at */gpfs/group/jopark/ref_genome/3M-february-2018.txt*

```bash
        /gpfs/group/jin/software/STAR-2.7.9a/source/STAR --runThreadN=16 \
            --genomeDir /gpfs/home/jopark/ref_genome  \
            --readFilesIn /dir/to/R2/fastq1,/dir/to/R2/fastq2,... /dir/to/R1/fastq1,/dir/to/R1/fastq2...\
            --soloType Droplet --soloCBwhitelist /dir/to/whitelist\
            --soloCellFilter EmptyDrops_CR \
            --soloBarcodeReadLength 28 \
--soloFeatures SJ  #if you want to output splice junction data
```

# Generating Index
*Note: once someone generates the index for a given species, different experiments can be aligned to the same index*

Index for mouse based on mm10 reference can be found at:
```bash
/gpfs/group/jin/ref_genomes/mm10
```

For new indexes
1. Download reference genome (from 10x or other sources)
- Reference should include the following two files:
	*GTF annotation files (.gtf)*
	*FASTA sequence files (.fa)*
	
2. Run STAR in genomeGenerate mode
- For sjdbOverhang parameter, use readLength-1
```bash
	/gpfs/group/jin/software/STAR-2.7.9a/source/STAR --runThreadN=16 \
		--runMode genomeGenerate --genomeDir /dir/to/output/index \
		--genomeFastaFiles /dir/to/genome/fasta \
		--sjdbGTFfile /dir/to/genome/gtf --sjdbOverhang 74 
		/gpfs/group/jin/software/STAR-2.7.9a/source/STAR --runThreadN 16 --runMode genomeGenerate --genomeDir /gpfs/group/jin/ref_genomes/mm10_GFP/ --genomeFastaFiles /gpfs/group/jin/ref_genomes/mm10_GFP/genome_GFP.fa --sjdbGTFfile /gpfs/group/jin/ref_genomes/mm10_GFP/genome_GFP.gtf --sjdbOverhang 99
```

# Quality control
After navigating to the directory where you want the QC report stored, run
```bash
multiqc /dir/with/alignment/outputs
```
Quality control output will be output as multiqc_report.html (see [here](https://docs.seqera.io/multiqc#using-multiqc) for more info)

# Alternative resources:
[Qualimap](http://qualimap.conesalab.org/#what-is-qualimap)
[Quality of RNA-seq Tool-Set](http://hartleys.github.io/QoRTs/)

# Downstream Analyses
Differential expression analysis using [DESeq2 ](https://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#htseq-count-input)
